phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4373,10 +4380,39 @@ private PhoenixConnection upgradeSystemSequence(
             } else {
                 nSequenceSaltBuckets = getSaltBuckets(e);
             }
+
+            updateSystemSequenceWithCacheOnWriteProps(metaConnection);
         }
         return metaConnection;
     }
 
+    private void updateSystemSequenceWithCacheOnWriteProps(PhoenixConnection metaConnection) throws
+        IOException, SQLException {
+
+        try (Admin admin = getAdmin()) {
+            PTable pTable = PhoenixRuntime.getTable(metaConnection,
+                PhoenixDatabaseMetaData.SYSTEM_SEQUENCE_NAME);
+            TableDescriptor oldTD = admin.getDescriptor(
+                SchemaUtil.getPhysicalTableName(PhoenixDatabaseMetaData.SYSTEM_SEQUENCE_NAME,
+                    metaConnection.getQueryServices().getProps()));
+            ColumnFamilyDescriptor oldCf = oldTD.getColumnFamily(
+                SchemaUtil.getEmptyColumnFamily(pTable));
```

Comment:
Since this table is one of the System tables, we can directly use `DEFAULT_COLUMN_FAMILY_BYTES` instead of calling `SchemaUtil#getEmptyColumnFamily`. Moreover, this would also help getting rid of calling `PhoenixRuntime#getTable`.

```
            TableDescriptor oldTD = admin.getDescriptor(
                SchemaUtil.getPhysicalTableName(PhoenixDatabaseMetaData.SYSTEM_SEQUENCE_NAME,
                    metaConnection.getQueryServices().getProps()));
            ColumnFamilyDescriptor oldCf = oldTD.getColumnFamily(QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES);
```

Reply:
Aah got it. My thought was to use the utility method, and avoid writing the underlying defaulting logic ourself. 

But i get it now :) avoiding the PhoenixRuntime is also a plus. Will make the change.